docs/markdown/concepts/polkadot-light-client/mmr-light-client.md

```diff
@@ -6,12 +6,15 @@ permalink: /concepts/polkadot-light-client-verifier/mmr-light-client
 parent: Polkadot Light Client Verifier
 grand_parent: Concepts and Architecture
 ---
+
 # Polkadot Relay Chain MMR Light Client
 
-After the interactive protocol runs, we have new [BEEFY](https://github.com/paritytech/grandpa-bridge-gadget) MMR commitments. These are merkle mountain ranges that contain data for updates to the Polkadot Validator set and data for new parachain headers.
+After the interactive protocol runs, we have new [BEEFY](https://github.com/paritytech/grandpa-bridge-gadget) MMR commitments. These are the root hashes of merkle mountain ranges that contain data for updates to the Polkadot Validator set and data for new relay chain headers.
 
 We use merkle proofs to verify the contents of each BEEFY MMR, extracting the above two kinds of data.
 
 New validator set updates are applied to our interactive protocol client so that it knows which validator signatures will be valid for the next block.
 
 The new Parachain header for our Snowbridge chain is verified and then provided to our [Parachain Light Client](/concepts/polkadot-light-client-verifier/parachain-light-client) to be used for verifying and processing new bridge messages.
+
+<!-- TODO: why do we need to store those headers? It should be cheaper to just store the MMR root and then verify a proof against that. -->
```

Comment:
hmm yeh, i think it depends on how things go in practice, like, if we are doing batch processing of parachain commitments, then we just can just do a single MMR verification to process a full commitment without storing the parachain/relay chain header. if we are doing individual processing of each message, for example in a merkle tree commitment, then we'd need to re-verify the MMR each time if we don't store it?